### ISSUE TITLE
Packet processing error when renaming an item in an anvil with color anvil patch.

### ISSUE THREAD

**Issue by mibby** [pl3xgaming/Purpur#176]
**Describe the bug**
```
[03:31:52] [Server thread/ERROR]: Encountered unexpected exception sending packets
[03:31:52] [Server thread/ERROR]: Error whilst processing packet net.minecraft.server.v1_16_R3.PacketPlayInItemName@371feb38 for mibby[/IP]
java.lang.StringIndexOutOfBoundsException: begin 2, end 8, length 2
	at java.lang.String.checkBoundsBeginEnd(String.java:3319) ~[?:?]
	at java.lang.String.substring(String.java:1874) ~[?:?]
	at net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializerImpl.decodeTextFormat(LegacyComponentSerializerImpl.java:152) ~[1.16.5-patched.jar]
	at net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializerImpl.deserialize(LegacyComponentSerializerImpl.java:217) ~[1.16.5-patched.jar]
	at net.minecraft.server.v1_16_R3.ContainerAnvil.e(ContainerAnvil.java:266) ~[1.16.5-patched.jar]
	at net.minecraft.server.v1_16_R3.ContainerAnvil.a(ContainerAnvil.java:338) ~[1.16.5-patched.jar]
	at net.minecraft.server.v1_16_R3.PlayerConnection.a(PlayerConnection.java:837) ~[1.16.5-patched.jar]
	at net.minecraft.server.v1_16_R3.PacketPlayInItemName.a(SourceFile:30) ~[1.16.5-patched.jar]
	at net.minecraft.server.v1_16_R3.PacketPlayInItemName.a(SourceFile:8) ~[1.16.5-patched.jar]
	at net.minecraft.server.v1_16_R3.PlayerConnectionUtils.lambda$ensureMainThread$1(PlayerConnectionUtils.java:43) ~[1.16.5-patched.jar]
	at net.minecraft.server.v1_16_R3.TickTask.run(SourceFile:18) ~[1.16.5-patched.jar]
	at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:136) ~[1.16.5-patched.jar]
	at net.minecraft.server.v1_16_R3.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[1.16.5-patched.jar]
	at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:109) ~[1.16.5-patched.jar]
	at net.minecraft.server.v1_16_R3.MinecraftServer.bb(MinecraftServer.java:1218) ~[1.16.5-patched.jar]
	at net.minecraft.server.v1_16_R3.MinecraftServer.executeNext(MinecraftServer.java:1211) ~[1.16.5-patched.jar]
	at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.awaitTasks(IAsyncTaskHandler.java:119) ~[1.16.5-patched.jar]
	at net.minecraft.server.v1_16_R3.MinecraftServer.sleepForTick(MinecraftServer.java:1187) ~[1.16.5-patched.jar]
	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1027) ~[1.16.5-patched.jar]
	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:176) ~[1.16.5-patched.jar]
	at java.lang.Thread.run(Thread.java:834) [?:?]
[03:31:52] [Server thread/INFO]: mibby lost connection: Packet processing error
```

**To Reproduce**
1. Enable ``allow-colors: true`` purpur setting.
2. Have permission for anvil color changing and ignoring italic.
3. Try to add a hex color by setting ``&r&#`` to the name of an item in an anvil.
4. Get kicked.

**Version**
Purpur dev 991

**Is bug present on Paper, Spigot, CraftBukkit, or Vanilla**
No.

**Additional context**
RE: https://github.com/pl3xgaming/Purpur/issues/153

For context, disabling Purpur's patch and using the ColoredAnvil plugin with patched HEX/forced disabled italic, does not have this problem. https://github.com/Kikisito/ColoredAnvils

Also, would it perhaps be possible to add an option to update the item automatically to remove italic unless specified with the formatting code? Instead of requiring players to use ``&r``. 

I also noticed if you put in a name and then ``ctrl + a`` delete all text in the text field, the preview item doesn't update / refresh to remove the text. This is probably a minecraft bug since it also occurs with ColoredAnvil.

**Comment by jpenilla** [comment]
This is a bug in the legacy component serializer which has been [fixed already](https://github.com/KyoriPowered/adventure/pull/282/files#diff-ebf36a421b9c6c0fbec205a53931b481daef475634e0c4a386cf7cc13fbbcff7R124), however the fix will be included in a future version of adventure. I will look into a temporary workaround for this issue in the meantime

**Comment by kashike** [comment]
[KyoriPowered/adventure@4.5.1](https://github.com/KyoriPowered/adventure/releases/tag/v4.5.1)

**Comment by jpenilla** [comment]
Fixed in 4602b9b2fc637d5d47fade76bbeaac01e83bd17b
